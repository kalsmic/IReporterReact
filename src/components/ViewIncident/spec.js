import { shallow } from 'enzyme';
import React from 'react';
import { mapDispatchToProps, mapStateToProps, ViewIncident } from './index';

describe('CreateIncident Component', () => {
  const viewIncidentFn = jest.fn();
  const props = {
    viewIncident: viewIncidentFn,
    error: '',
    message: '',
    incident: {},
  };
  const wrapper = shallow(<ViewIncident {...props} />);

  it('should render without crushing', () => {
    expect(wrapper)
      .toMatchSnapshot();
  });

  it('should map state to props', () => {
    const mockedState = {
      incidentReducer: {
        message: 'Created Successfully',
        error: '',
        incident: {},
      },
    };
    const state = mapStateToProps(mockedState);
    expect(state.message)
      .toBe('Created Successfully');
    expect(state.error)
      .toBe('');
  });

  it('should map dispatch to props', () => {
    const mockedDispatch = jest.fn();

    mapDispatchToProps(mockedDispatch)
      .viewIncident();
    expect(mockedDispatch)
      .toHaveBeenCalled();
  });
});
