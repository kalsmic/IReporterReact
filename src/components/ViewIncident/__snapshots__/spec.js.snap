// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateIncident Component should render without crushing 1`] = `
<Root>
  <h3
    className="text-blue bd_line_bottom"
  >
    View Incident Record Details
  </h3>
  <div
    className="incident__card"
  />
  <div
    className="incident__card__header"
  >
    <h1
      className="incident__card__header__title"
    />
    <span
      className="incident__card__header__created_on"
    >
      Created On:
       
    </span>
  </div>
  <div
    className="incident__card-comment"
  />
  <div
    className="incident__card-map "
  />
  <div
    className="incident__card__footer"
  >
    <span
      className="incident__card-status"
    >
      status:
       
       
    </span>
    <span>
      Created by :
       
    </span>
  </div>
</Root>
`;
